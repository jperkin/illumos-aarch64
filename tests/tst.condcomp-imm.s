/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2017 Joyent, Inc.
 */

/*
 * C3.5.4 Conditional compare (immediate)
 */
.macro condcomp_imm op
	/* 32-bit */
	\op	w0,  #0,  #0,  eq
	\op	w0,  #31, #1,  ne
	\op	wzr, #0,  #2,  cs
	\op	wzr, #31, #3,  cc
	\op	w0,  #0,  #4,  mi
	\op	w0,  #31, #5,  pl
	\op	wzr, #0,  #6,  vs
	\op	wzr, #31, #7,  vc
	\op	w0,  #0,  #8,  hi
	\op	w0,  #31, #9,  ls
	\op	wzr, #0,  #10, ge
	\op	wzr, #31, #11, lt
	\op	w0,  #0,  #12, gt
	\op	w0,  #31, #13, le
	\op	wzr, #0,  #14, al
	\op	wzr, #31, #15, nv
	/* 64-bit */
	\op	x0,  #0,  #0,  eq
	\op	x0,  #31, #1,  ne
	\op	xzr, #0,  #2,  cs
	\op	xzr, #31, #3,  cc
	\op	x0,  #0,  #4,  mi
	\op	x0,  #31, #5,  pl
	\op	xzr, #0,  #6,  vs
	\op	xzr, #31, #7,  vc
	\op	x0,  #0,  #8,  hi
	\op	x0,  #31, #9,  ls
	\op	xzr, #0,  #10, ge
	\op	xzr, #31, #11, lt
	\op	x0,  #0,  #12, gt
	\op	x0,  #31, #13, le
	\op	xzr, #0,  #14, al
	\op	xzr, #31, #15, nv
.endm

.text
.globl libdis_test
.type libdis_test, @function
libdis_test:
	/*
	 * C5.6.32 CCMN (immediate)
	 * C5.6.34 CCMP (immediate)
	 */
	condcomp_imm ccmn
	condcomp_imm ccmp
.size libdis_test, [.-libdis_test]
