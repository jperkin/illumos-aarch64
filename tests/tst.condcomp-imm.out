    libdis_test:       00 08 40 3a  ccmn w0, #0x0, #0x0, eq
    libdis_test+0x4:   01 18 5f 3a  ccmn w0, #0x1f, #0x1, ne
    libdis_test+0x8:   e2 2b 40 3a  ccmn wzr, #0x0, #0x2, cs
    libdis_test+0xc:   e3 3b 5f 3a  ccmn wzr, #0x1f, #0x3, cc
    libdis_test+0x10:  04 48 40 3a  ccmn w0, #0x0, #0x4, mi
    libdis_test+0x14:  05 58 5f 3a  ccmn w0, #0x1f, #0x5, pl
    libdis_test+0x18:  e6 6b 40 3a  ccmn wzr, #0x0, #0x6, vs
    libdis_test+0x1c:  e7 7b 5f 3a  ccmn wzr, #0x1f, #0x7, vc
    libdis_test+0x20:  08 88 40 3a  ccmn w0, #0x0, #0x8, hi
    libdis_test+0x24:  09 98 5f 3a  ccmn w0, #0x1f, #0x9, ls
    libdis_test+0x28:  ea ab 40 3a  ccmn wzr, #0x0, #0xa, ge
    libdis_test+0x2c:  eb bb 5f 3a  ccmn wzr, #0x1f, #0xb, lt
    libdis_test+0x30:  0c c8 40 3a  ccmn w0, #0x0, #0xc, gt
    libdis_test+0x34:  0d d8 5f 3a  ccmn w0, #0x1f, #0xd, le
    libdis_test+0x38:  ee eb 40 3a  ccmn wzr, #0x0, #0xe, al
    libdis_test+0x3c:  ef fb 5f 3a  ccmn wzr, #0x1f, #0xf, nv
    libdis_test+0x40:  00 08 40 ba  ccmn x0, #0x0, #0x0, eq
    libdis_test+0x44:  01 18 5f ba  ccmn x0, #0x1f, #0x1, ne
    libdis_test+0x48:  e2 2b 40 ba  ccmn xzr, #0x0, #0x2, cs
    libdis_test+0x4c:  e3 3b 5f ba  ccmn xzr, #0x1f, #0x3, cc
    libdis_test+0x50:  04 48 40 ba  ccmn x0, #0x0, #0x4, mi
    libdis_test+0x54:  05 58 5f ba  ccmn x0, #0x1f, #0x5, pl
    libdis_test+0x58:  e6 6b 40 ba  ccmn xzr, #0x0, #0x6, vs
    libdis_test+0x5c:  e7 7b 5f ba  ccmn xzr, #0x1f, #0x7, vc
    libdis_test+0x60:  08 88 40 ba  ccmn x0, #0x0, #0x8, hi
    libdis_test+0x64:  09 98 5f ba  ccmn x0, #0x1f, #0x9, ls
    libdis_test+0x68:  ea ab 40 ba  ccmn xzr, #0x0, #0xa, ge
    libdis_test+0x6c:  eb bb 5f ba  ccmn xzr, #0x1f, #0xb, lt
    libdis_test+0x70:  0c c8 40 ba  ccmn x0, #0x0, #0xc, gt
    libdis_test+0x74:  0d d8 5f ba  ccmn x0, #0x1f, #0xd, le
    libdis_test+0x78:  ee eb 40 ba  ccmn xzr, #0x0, #0xe, al
    libdis_test+0x7c:  ef fb 5f ba  ccmn xzr, #0x1f, #0xf, nv
    libdis_test+0x80:  00 08 40 7a  ccmp w0, #0x0, #0x0, eq
    libdis_test+0x84:  01 18 5f 7a  ccmp w0, #0x1f, #0x1, ne
    libdis_test+0x88:  e2 2b 40 7a  ccmp wzr, #0x0, #0x2, cs
    libdis_test+0x8c:  e3 3b 5f 7a  ccmp wzr, #0x1f, #0x3, cc
    libdis_test+0x90:  04 48 40 7a  ccmp w0, #0x0, #0x4, mi
    libdis_test+0x94:  05 58 5f 7a  ccmp w0, #0x1f, #0x5, pl
    libdis_test+0x98:  e6 6b 40 7a  ccmp wzr, #0x0, #0x6, vs
    libdis_test+0x9c:  e7 7b 5f 7a  ccmp wzr, #0x1f, #0x7, vc
    libdis_test+0xa0:  08 88 40 7a  ccmp w0, #0x0, #0x8, hi
    libdis_test+0xa4:  09 98 5f 7a  ccmp w0, #0x1f, #0x9, ls
    libdis_test+0xa8:  ea ab 40 7a  ccmp wzr, #0x0, #0xa, ge
    libdis_test+0xac:  eb bb 5f 7a  ccmp wzr, #0x1f, #0xb, lt
    libdis_test+0xb0:  0c c8 40 7a  ccmp w0, #0x0, #0xc, gt
    libdis_test+0xb4:  0d d8 5f 7a  ccmp w0, #0x1f, #0xd, le
    libdis_test+0xb8:  ee eb 40 7a  ccmp wzr, #0x0, #0xe, al
    libdis_test+0xbc:  ef fb 5f 7a  ccmp wzr, #0x1f, #0xf, nv
    libdis_test+0xc0:  00 08 40 fa  ccmp x0, #0x0, #0x0, eq
    libdis_test+0xc4:  01 18 5f fa  ccmp x0, #0x1f, #0x1, ne
    libdis_test+0xc8:  e2 2b 40 fa  ccmp xzr, #0x0, #0x2, cs
    libdis_test+0xcc:  e3 3b 5f fa  ccmp xzr, #0x1f, #0x3, cc
    libdis_test+0xd0:  04 48 40 fa  ccmp x0, #0x0, #0x4, mi
    libdis_test+0xd4:  05 58 5f fa  ccmp x0, #0x1f, #0x5, pl
    libdis_test+0xd8:  e6 6b 40 fa  ccmp xzr, #0x0, #0x6, vs
    libdis_test+0xdc:  e7 7b 5f fa  ccmp xzr, #0x1f, #0x7, vc
    libdis_test+0xe0:  08 88 40 fa  ccmp x0, #0x0, #0x8, hi
    libdis_test+0xe4:  09 98 5f fa  ccmp x0, #0x1f, #0x9, ls
    libdis_test+0xe8:  ea ab 40 fa  ccmp xzr, #0x0, #0xa, ge
    libdis_test+0xec:  eb bb 5f fa  ccmp xzr, #0x1f, #0xb, lt
    libdis_test+0xf0:  0c c8 40 fa  ccmp x0, #0x0, #0xc, gt
    libdis_test+0xf4:  0d d8 5f fa  ccmp x0, #0x1f, #0xd, le
    libdis_test+0xf8:  ee eb 40 fa  ccmp xzr, #0x0, #0xe, al
    libdis_test+0xfc:  ef fb 5f fa  ccmp xzr, #0x1f, #0xf, nv
